### PR TITLE
fix ci + rustfmt

### PR DESCRIPTION
**Pull request by peddermaster2**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,10 @@ language: rust
 rust: nightly
 before_script:
   - export PATH="$PATH:$HOME/.cargo/bin"
-  - which rustfmt || cargo install --vers 0.8.4 rustfmt
+  - rustup component add rustfmt-preview
 
 script:
-    - cargo fmt -- --write-mode=diff
```
Comment (illegalprime) [inline .travis.yml]: i like this

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Rust-WebSocket [![Build Status](https://travis-ci.org/websocket-rs/rust-websocket.svg?branch=master)](https://travis-ci.org/websocket-rs/rust-websocket)
```
Comment (illegalprime) [inline README.md]: this is also very good

File: .rustfmt.toml [hunk 3]
```diff
@@ -1,8 +1 @@
 hard_tabs = true
-array_layout = "Block"
```
Comment (illegalprime) [inline .rustfmt.toml]: Could you keep these rules and do this PR, then do another PR about changing the project's style?

Comment (peddermaster2) [reply .rustfmt.toml]: I removed them becuase they no longer have any use. I have the console output with all this lines active:
```console
$ cargo fmt
Warning: Unknown configuration option `array_layout`
Warning: Unknown configuration option `chain_indent`
Warning: Unknown configuration option `chain_one_line_max`
Warning: Unknown configuration option `fn_args_layout`
Warning: Unknown configuration option `take_source_hints`
Warning: Unknown configuration option `write_mode`
```

**Comment by illegalprime** [comment]
OK, I have some time today so I'll merge this and try to bring back all the PRs that'll be affected.
